components/storage/blockdevice/COMPONENT_SD/SDBlockDevice.cpp

```diff
@@ -1129,29 +1129,22 @@ void SDBlockDevice::_spi_wait(uint8_t count)
 
 void SDBlockDevice::_spi_init()
 {
-    _spi.lock();
```

Comment:
Why are the locks removed in here?

Reply:
lock and unlock is made in select and unselect.

Reply:
but __select_ and __deselect_ are not called here.

Reply:
`lock` and `unlock` are called separately in each `frequency`, `format` and `set_default_write_value` -function.

Reply:
@VeijoPesonen exactly Jarno is right.

Reply:
Ok, so there is no reason to protect the sequence somehow. Which raises a question that should _SPI.h_ expose those lock/unlock-functions if all the public functions call those implicitly? @kjbracey-arm any thoughts.

Reply:
Arguably lock/unlock shouldn't be public API any more.  Most driver classes have virtual protected lock/unlock that lets you override them for a `UnlockedX` class or whatever, but for this class they were then made public to let apps use them as a a bus lock.

`select`/`deselect` basically replace them for app use. But there's no way to deprecate `lock` as a public method while keeping it available for the override purpose.

Now, you don't normally need to manually call `select` or `deselect` either, but you do need to call them if you want to group multiple transactions together.

An analogue in POSIX is `flock`/`funlock` - each stdio call like `printf` does call them internally, but you'd want to call them yourself if you wanted to keep two `printf`s together.

So these two successfully lock against each other (X can't occur between A and B).

     // Task 1
     spi.write(X); 

     // Task 2
     spi.select();
     spi.write(A);
     spi.write(B);
     spi.deselect();

But only because X has the implicit select.

(Don't make the mistake I made a few years ago of writing code like that when I had a bus for which a single op DIDN'T have a built-in implicit mutex lock. If it didn't you'd need to put the manual select around the single op too. I got confused between the "lock two things" together function and the "mutex against other tasks" function).